webapp/src/main/java/com/box/l10n/mojito/monitoring/QuartzPendingJobsReportingTask.java

```diff
@@ -21,28 +25,40 @@
 
     private JdbcTemplate jdbcTemplate;
     private MeterRegistry meterRegistry;
+    private Map<PendingJob, AtomicLong> queueSizes;
 
     public QuartzPendingJobsReportingTask(@Autowired DataSource dataSource,
                                           @Autowired MeterRegistry meterRegistry) {
         this.meterRegistry = meterRegistry;
         this.jdbcTemplate = new JdbcTemplate(dataSource);
+        this.queueSizes = new ConcurrentHashMap<>();
     }
 
     @Scheduled(fixedRateString = "${l10n.management.metrics.quartz.sql-queue-monitoring.execution-rate}")
     public void reportPendingJobs() {
-        fetchResults().forEach(this::reportResults);
+        // First we remove all of our previously stored results (notice line 52). If a job clears from the queue
+        // we would still report it as having pending jobs as it'll retrieve its size from our queueSizes map.
+        queueSizes.clear();
```

Comment:
I guess that makes sense for ephemeral jobs but isn't that clearing the gauge for some jobs that shouldn't be cleared?

Reply:
Sure, look at line 48 where we reinsert values into the map all over again. And I've covered that same test case you mention in the tests. 

Also, if we were to remove just those that aren't returning results during a particular measurement, we'll need to maintain a record of previously obtained results to compare. Given this is returning 20 rows from the table approximately, is just easier to clear the map.